src/main/java/org/testng/internal/Invoker.java

```diff
@@ -1216,6 +1216,7 @@ private void handleInvocationResults(ITestNGMethod testMethod,
         failure.instances.add(instance);
       }
       testResult.setStatus(ITestResult.SKIP);
+      testResult.setWasRetried(true);
```

Comment:
Why not having a new status?
According to the user feedbacks on issues, I think it is what they'd like.

Reply:
Sure, but a major release is the perfect moment to change it ;)

Reply:
Now am curious. Can you please let me know as to what is the drawback with the current implementation ?

Reply:
The current implementation could be good but I'm not sure to understand the logic of the semantics. 
What does mean if wasRetried is true and the status is SUCCESS?
In fact, maybe it should be the case when invocation-count is used.

And skip is clearly not a good status but failed had some problems too (maybe we can fix them now with `wasRetried`).

IMO, we can go a bit further with this fix.

Reply:
> What does mean if wasRetried is true and the status is SUCCESS?

TestNG currently maintains the intermediate results of every retried iterations before they get determined to be success or failed. So an ITestResult may have its status as `SKIP` and `wasRetried`, but it will never have `SUCCESS` and `wasRetried`.

> And skip is clearly not a good status but failed had some problems too (maybe we can fix them now with wasRetried).

Sure. But I based my changes based on historic discussions so that I can avoid repeated re-work on pull requests. The previous discussions on this to me suggested that we weren't too inclined to have a new status. Moreover, I dont know what sort of changes would be required at the plugins for IDEs (IntelliJ and eclipse I know of), to accommodate and distinguish between the new status. 
Which is why I decided to take this path of not affecting any of the existing code paths. We can enhance our documentation to tell people that `SKIP` + `wasRetried` is not the same as `SKIP` (if its a question of educating and clarifying ambiguities to our users).

> IMO, we can go a bit further with this fix.

Sure we can. But please help me by calling them out. Currently I have made the changes which I felt were needed to address the ask of this issue. (Yes enhancing our default inbuilt reports to accommodate this change would be a different PR).

@cbeust - Since you had expressed concerns earlier that you were not in for a new status (https://github.com/cbeust/testng/issues/878#issuecomment-158120731) , please help comment and let us know what direction we should be taking.